File: doc-examples/jdbc-example-java/src/main/java/example/StudentView.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.JsonView;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;
import io.micronaut.data.annotation.sql.JoinColumn;
import java.util.List;

// tag::class-example[]
@JsonView(entity = Student.class)
public class StudentView {
    @Id
    @GeneratedValue(GeneratedValue.Type.IDENTITY)
    private Long id;

    private String name;

    @JoinColumn(name = "id", referencedColumnName = "student_id")
    @Relation(Relation.Kind.ONE_TO_MANY)
    private List<StudentScheduleSubView> schedule;
}
// end::class-example[]

File: doc-examples/jdbc-example-python/src/main/python/example/StudentView.py
from dataclasses import dataclass, field
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, JsonView, Relation
from micronaut.data.annotation.sql import JoinColumn

from example.Student import Student
from example.StudentScheduleSubView import StudentScheduleSubView


# tag::class-example[]
@JsonView(entity=Student)
@dataclass
class StudentView:
    name: str
    schedule: Annotated[list[StudentScheduleSubView], JoinColumn(name="id", referencedColumnName="student_id"), Relation("ONE_TO_MANY")] = field(default_factory=list)
    id: Annotated[int | None, Id, GeneratedValue("IDENTITY")] = None
# end::class-example[]
